### PR TITLE
zero:request:2.0.0

### PR DESCRIPTION
**Pull request by MadroneXBT**
Note:
-  The contract interface in encodeVaultTransferRequestLoan and encodeVaultTransferRequestRepay were updated based on d1ll0n's changes to the vault contract here: https://github.com/zerodao-finance/zero-protocol/blob/wip-vault/contracts/erc4626/vault/ZeroBTCLoans.sol

### CODE REVIEW

File: packages/watcher/lib/watcher.js [hunk 1]
```diff
@@ -6,19 +6,17 @@ const { UnderwriterTransferRequest } = require("zero-protocol/dist/lib/zero");
...
 const encodeTransferRequestRepay = (transferRequest, queryResult) => {
   const contractInterface = new ethers.utils.Interface([
-    "function repay(address, address, address, uint256, uint256, uint256, address, bytes32, bytes, bytes)",
+    "function repay(address, address, uint256, uint256, address, bytes32, bytes, bytes)",
   ]);
   return contractInterface.encodeFunctionData("repay", [
     transferRequest.underwriter,
     transferRequest.destination(),
-    transferRequest.asset,
     transferRequest.amount,
```
Comment (MadroneXBT) [inline packages/watcher/lib/watcher.js:14]: What's the difference b/w transferRequest.amount & queryResult.amount?

File: packages/watcher/lib/watcher.js [hunk 2]
```diff
@@ -6,19 +6,17 @@ const { UnderwriterTransferRequest } = require("zero-protocol/dist/lib/zero");
...
-    queryResult.amount,
```
Comment (MadroneXBT) [inline packages/watcher/lib/watcher.js:16]: What's the difference b/w transferRequest.amount & queryResult.amount?

**Comment by MadroneXBT** [comment]
@kungfuflex The encodeControllerTransferRequestLoan and encodeControllerTransferRequestRepay logic now remains the same. I just renamed the fn to be distinct.

**Review by kungfuflex — COMMENTED** [review]
The section in lib/pending.js needs to be deleted that is !CONTROLLER_DEPLOYMENTS. The code was intended to only send a loan() transaction if it were a contract that was not one of the controllers, which do not have a loan() function. But, we already validate that the contract address supports loan() when we check VAULT_DEPLOYMENTS, so the following block solves this problem

The section in lib/watcher.js should check CONTROLLER_DEPLOYMENTS not !CONTROLLER_DEPLOYMENTS to verify that it is supposed to encode a controller repay

**Review by kungfuflex — APPROVED** [review]
Looks good!

**Pull request merged**